File: backend_golang/presentation/handler/create_todo_handler.go
package handler

import (
	"backend_golang/application_service/usecase"
	"net/http"

	"github.com/google/uuid"
	"github.com/labstack/echo/v4"
)

type CreateTodoHandler struct {
	cu usecase.CreateTodoUsecase
}

func NewCreateTodoHandler(cu usecase.CreateTodoUsecase) HTTPHandler {
	return &CreateTodoHandler{
		cu: cu,
	}
}

type CreateTodoRequest struct {
	Title       string  `json:"title" validate:"required"`
	Description *string `json:"description"`
}

type CreateTodoResponse struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description *string   `json:"description"`
	Completed   bool      `json:"completed"`
}

func (h *CreateTodoHandler) Handle(c echo.Context) error {
	var req CreateTodoRequest
	if err := c.Bind(&req); err != nil {
		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
	}
	if err := c.Validate(&req); err != nil {
		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
	}
	todo, err := h.cu.Handle(c.Request().Context(), usecase.CreateTodoUsecaseInput{
		Title:       req.Title,
		Description: req.Description,
	})
	if err != nil {
		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
	}
	t := todo.Serialize()

	return c.JSON(http.StatusCreated, CreateTodoResponse{
		ID:          t.ID,
		Title:       t.Title,
		Description: t.Description,
		Completed:   t.Completed,
	})
}

File: backend_python/src/todo_api/presentation/handler/create_todo_handler.py
from pydantic import BaseModel

from todo_api.application_service.usecase.create_todo_usecase import (
    CreateTodoUsecase,
    CreateTodoUsecaseInput,
)


class CreateTodoRequest(BaseModel):
    title: str
    description: str | None = None


class CreateTodoResponse(BaseModel):
    id: str
    title: str
    description: str | None = None
    completed: bool


class CreateTodoHandler:
    def __init__(self, create_todo_usecase: CreateTodoUsecase) -> None:
        self.create_todo_usecase = create_todo_usecase

    def handle(self, request: CreateTodoRequest) -> CreateTodoResponse:
        result = self.create_todo_usecase.execute(
            CreateTodoUsecaseInput(
                title=request.title,
                description=request.description,
            )
        )

        return CreateTodoResponse(
            id=str(result.todo.id),
            title=result.todo.title,
            description=result.todo.description,
            completed=result.todo.completed,
        )
